datasource/src/main/scala/quasar/physical/s3/AsyncHttpClientBuilder.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2014–2019 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.physical.s3
+
+import slamdata.Predef._
+
+import quasar.contrib.proxy.Search
+
+import org.asynchttpclient.proxy.{ProxyServer, ProxyServerSelector}
+import org.asynchttpclient.uri.Uri
+import org.asynchttpclient.{AsyncHttpClientConfig, DefaultAsyncHttpClientConfig}
+
+import org.http4s.client.Client
+import org.http4s.client.asynchttpclient.AsyncHttpClient
+import org.http4s.util.threads.threadFactory
+
+import java.net.{InetSocketAddress, ProxySelector}
+
+import scala.concurrent.ExecutionContext
+import scala.collection.JavaConverters._
+
+import cats.effect.{ConcurrentEffect, Resource}
+
+object AsyncHttpClientBuilder extends Http4sClientBuilder {
```

Comment:
Copied over from url datasource. In a next PR we could move it to `quasar.contrib` and deduplicate.